### DISCUSSION TITLE
General Test Ideas

### DISCUSSION THREAD

**Discussion by TheENGR** [CS-6381/FinalProject#5] [Ideas]
Hey all, lets come up with some basic ideas for tests!

**Comment by TheENGR** [comment]
1) How many steps to install?
2) What OSes does it run on?
    1) How much RAM do you need? 
3) How configurable is it?

**Reply by rich-java-dev** [reply]
- strengths
- weaknesses
- Simply cannot handle certain tasks/not designed for

**Reply by jdunn-git** [reply]
4. Scalablility (both vertical and horizontal, if applicable)
5. Fault tolerability
6. Real world use case(s)

**Comment by TheENGR** [comment]
OSes to Test:

1) Windows 10
2) Ububtu 20.04 (LTS)
3) Mac? 
4) Docker (Ububtu 20.04)

**Reply by rich-java-dev** [reply]
for this, I almost want to assume we can work exclusively on Ubuntu 20.04, in that most of these open source tools are designed to be ran on linux vs windows. Just my 2 cents tho.

**Reply by laynemoseley** [reply]
I agree with Rich on this point. Support for this software on Mac/Windows is generally only for development. When it is run in production in the cloud, it's safe to assume that it'll be on Linux.
